src/Field/Choices/StoryChoices.php

```diff
@@ -12,10 +13,12 @@ final class StoryChoices implements ChoicesInterface
 {
 	/**
 	 * @param array<string>|ComponentsWithTags $restrictContentTypes
+	 * @param string|\BackedEnum|null          $referencedStoryDataMode This is a Key that will be used by the corresponding StoryNormalizer to find out which data from the referenced Story is needed by the component
 	 */
 	public function __construct (
 		private readonly array|ComponentsWithTags $restrictContentTypes = [],
 		private readonly string $restrictToPath = "",
+		private readonly string|\BackedEnum|null $referencedStoryDataMode = null,
```

Comment:
This should not be set here, but instead in the normalizer's context.

Reply:
This is here so that it can be passed to the normalizer's context within the context. Otherwise we have no easy way to do this, or am I missing something?

Reply:
Your component would set the context when doing the recursive call in the normalizer. As only your component knows what data it needs.

Reply:
Ah, you want to set the context in the definition instead of the normalizer. We could do that, but then I would rename this variable.